src/core/service.c

```diff
@@ -2579,11 +2579,12 @@ static void service_notify_cgroup_empty_event(Unit *u) {
                  * SIGCHLD for. */
 
         case SERVICE_START:
-        case SERVICE_START_POST:
                 if (s->type == SERVICE_NOTIFY)
                         /* No chance of getting a ready notification anymore */
                         service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_PROTOCOL);
-                else if (s->pid_file_pathspec) {
+                else /* Fall through */
```

Comment:
No problem.
Do we want this change at all? It is clear that the test should apply only to SERVICE_START, yet as far as I can tell it is impossible for Type=notify services to end up here while in a SERVICE_START_POST state. The main reason for this change is, I think, that it is more 'correct'.